### PR TITLE
Fetch nearby vet centers

### LINKED ISSUE #27870: Automate "Nearby Vet Centers"
## Issue Description

As A Veteran, I want to know what other Vet Center facilities might be closer to me so that I can visit the one which is the most convenient. 

## Background: 

Vet Centers cover a catchment area. If you are a Veteran living at an edge of a catchment area, the closest location might actually be a satellite location of another Vet Center. 

For example: If I am a Veteran living southwest of Cincinnati, my closest Vet Center could be in Kentucky (especially if they have a satellite location on that side of their area. 

Currently, field staff at Vet Centers designate which locations should be displayed as "Nearby Vet Centers" but many are listing only the ones they are familiar with or none at all.
 
<details>
<summary > screenshot of current page </summary>

image

</details>

---
If a Vet Center has not entered any nearby locations, the section header will not display. This functionality should be maintained. 

<details>
<summary> Screenshot of San Luis Obispo, which has no nearby locations entered. </summary>

image

</details>

 ---

How might we automate this list?
(One idea: select Vet Centers/outstations within 80 miles? radius from the facilities API and remove the ones that are already posted in the local/satellite. And then show the first five sorted by distance.) 

## Dev Notes
- Consider a future state which might set the default of 80 allowing a radius to be changed by the local editor

---
## Tasks
- [x] Ping API for Vet Centers and outstations within 80 miles
- [x] Exclude locations already listed as Satellite or associated locations
- [x] Return the closest locations, up to 5
- [x] Order by distance
- [ ] Add tests
- [x] Add a feature flag to support testing in staging

## Acceptance Criteria
- [x] Up to 5 nearby Vet Center locations will be displayed, in ascending order
- [x] If the VC’s specified in Drupal aren’t included in the Lighthouse search, we will not display them
- [x] In the case where there are no locations at that radius, do not show the header
- [ ] Do not generate empty `` tags for vet centers fetched from Lighthouse.

### PR DESCRIPTION
**Pull request by mpelzsherman**
Resolves https://github.com/department-of-veterans-affairs/va.gov-team/issues/27870

Major refactor of nearByVetCenters.jsx.

Requires changes for https://github.com/department-of-veterans-affairs/content-build/pull/545 (merged)

### CODE REVIEW

File: src/applications/static-pages/facilities/vet-center/nearByVetCenters.jsx [hunk 1]
```diff
@@ -4,32 +4,54 @@ import LoadingIndicator from '@department-of-veterans-affairs/component-library/
...
+    } else {
+      fetchUnpublishedVetCenters();
+    }
+  }, []);
 
-  if (
-    props.vetCenters.some(v => !v.entity.entityPublished) &&
-    vetCenterFacilities.length === 0
-  ) {
+  if (props.loading) {
```
Comment (gescobardev) [inline src/applications/static-pages/facilities/vet-center/nearByVetCenters.jsx:50]: glad we have `props.loading`
looks much cleaner!

**Review by gescobardev — APPROVED** [review]

**Review by k80bowman — APPROVED** [review]

File: src/applications/static-pages/facilities/vet-center/NearByVetCenters.jsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+  if (props.facilitiesLoading) {
+    return <LoadingIndicator message="Loading facilities..." />;
+  }
+
+  // TODO: consider moving to a separate component
+  const renderVetCenter = (vetCenter, mainVetCenterPhone) => {
+    // TODO - fix unique key warning
+    // Tried adding this to the div but it causes rendering to fail for some reason:
+    // key={vetCenter.id}
```
Comment (Rleahy22) [inline src/applications/static-pages/facilities/vet-center/NearByVetCenters.jsx:85]: Could this be `key={vetCenter.id || vetCenter.fieldFacilityLocatorApiId}` or even `vetCenter.title.replaceAll(/\s/, '-')`? I'm not sure publishedVetCenters are guaranteed ids.

**Review by Rleahy22 — APPROVED** [review]
LGTM, but omitting keys in a list always worries me.

File: src/applications/static-pages/facilities/vet-center/NearByVetCenters.jsx (hunk 2 above)
Comment (mpelzsherman) [reply src/applications/static-pages/facilities/vet-center/NearByVetCenters.jsx:85]: Thanks Ryan - that worked! You da man.

**Pull request merged**